models/sharky/solana/sharky_solana_events.sql

```diff
@@ -0,0 +1,119 @@
+{{ config(
+    alias = 'events',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'tx_hash', 'id'],
+    post_hook='{{ expose_spells(\'["solana"]\',
+                                    "project",
+                                    "sharky",
+                                    \'["ennnas", "hosuke"]\') }}'
+    )
+}}
+
+{%- set project_start_date = '2023-01-14' %} -- TODO change back to '2022-04-14'
+{%- set sharky_smart_contract = 'SHARKobtfF1bHhxD2eqftjHBdVSCbKo9JtgK71FhELP' %}
+
+WITH sharky_txs AS (
+        SELECT tx_id AS id,
+               block_time
+        FROM {{ source('solana', 'account_activity') }}
```

Comment:
@ennnas one last attempt at trying to simplify this query, as we're aware solana data volume is massive :) for this table -- is it absolutely necessary to pull data from here and filter on shark contract address? is there anyway to only read from `solana.transactions` and filter on the contract address there? that way we avoid a join between larger tables.

if you see the magiceden spell [here](https://github.com/duneanalytics/spellbook/blob/main/models/magiceden/solana/magiceden_solana_events.sql#L123), it filters on values this way. is that applicable here?

Reply:
if not -- all good! it appears that account activity table when filtered down is ~1.7mil rows for all historical, which actually isn't too bad, but we may as well do one last sanity check

Reply:
@jeff-dude In the first implementation of the spell I was actually using the same approach of the magiceden spell, using the transactions table and filtering using `array_contains(account_keys, 'PROGRAM_ID')` but looking at the results from the CI it was actually slower that using the `account_activity` table to find the transactions of interest. Unfortunately I don't have a way to properly benchmark to make sure the results are "real" and not due to some external factors

Reply:
gotcha, good to know. thank you for the feedback. with that in mind, i'm comfortable with this approach as-is.